casepro/backend/junebug.py

```diff
@@ -40,6 +46,60 @@ def get_addresses(self, uuid):
         return (
             a['address'] for a in addresses if a.get('address') is not None)
 
+    def get_identities(self, **kwargs):
+        '''Get the list of identities filtered by the given kwargs.'''
+        url = '%s/api/v1/identities/search/?' % (self.base_url)
+        for key, value in kwargs.iteritems():
+            url = '%s%s=%s&' % (url, key, value)
+
+        identities = self.get_paginated_response(
+            url, params={'default': True})
+
+        return (
+            IdentityStoreContact(i) for i in identities if
+            i.get('details').get('name') is not "removed"
+        )
+
+
+class IdentityStoreContact(object):
+    """
+    Holds identity data for syncing
+    """
+    def __init__(self, json_data):
+        self.id = json_data.get('id')
+        self.version = json_data.get('version')
+        self.name = json_data.get('details').get('name')
+        self.addresses = json_data.get('details').get('addresses')
+        self.communicate_through = json_data.get('communicate_through')
+        self.operator = json_data.get('operator')
+        self.created_at = json_data.get('created_at')
+        self.created_by = json_data.get('created_by')
+        self.updated_at = json_data.get('updated_at')
+        self.updated_by = json_data.get('updated_by')
+
+
+class IdentityStoreContactSyncer(BaseSyncer):
+    """
+    Syncer for contacts from the Identity Store
+    """
+    model = Contact
+    remote_id_attr = 'id'
+
+    def local_kwargs(self, org, remote):
+        return {
+            'org': org,
+            'uuid': remote.id,
+            'name': remote.name,
+            'is_stub': False,
+        }
+
+    def update_required(self, local, remote, remote_as_kwargs):
+        if local.is_stub or local.name != remote.name:
+            return True
+
+        return not is_dict_equal(
+            local.get_fields, remote.fields, ignore_none_values=True)
```

Comment:
This should look almost exactly like the one in the rapidpro backend (except for perhaps a few minor changes in names of fields, etc). Frankly I'm a bit surprised this is needed (because in theory one can implement it entirely generically in terms of `local_kwargs`.